src/claim_proofs/ffi_wrapper/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+//! The `claim_proofs_ffi` is the Foreign Function Interface (FFI)
+//! for the `claim_proofs` library. It contains API for generating
+//! claim proofs and verifying them as part of the
+//! Asset Granularity Unique Identity project.
+
+extern crate libc;
+use libc::size_t;
+use std::slice;
+
+use cryptography::claim_proofs::{build_scope_claim_proof_data, compute_cdd_id, compute_scope_id};
+
+pub type ScopeClaimData = cryptography::claim_proofs::ScopeClaimData;
+pub type ScopeClaimProofData = cryptography::claim_proofs::ScopeClaimProofData;
+pub type ProofPublicKey = cryptography::claim_proofs::ProofPublicKey;
+pub type ProofKeyPair = cryptography::claim_proofs::ProofKeyPair;
+pub type CDDClaimData = cryptography::claim_proofs::CDDClaimData;
+pub type Signature = schnorrkel::Signature;
+pub type Scalar = curve25519_dalek::scalar::Scalar;
+pub type RistrettoPoint = curve25519_dalek::ristretto::RistrettoPoint;
+
+fn box_alloc<T>(x: T) -> *mut T {
+    Box::into_raw(Box::new(x))
+}
+
+// ------------------------------------------------------------------------
+// Data Structures
+// ------------------------------------------------------------------------
+
+/// Create a `Scalar` object from a 32 bytes array.
+///
+/// Caller is responsible for calling `scalar_free()` to deallocate this object.
+/// SAFETY: Caller is also responsible for making sure the `scalar_bits` pointer
+///         is not Null and points to a 32 bytes block of memory.
+#[no_mangle]
+pub unsafe extern "C" fn scalar_new(scalar_bits: *const u8) -> *mut Scalar {
+    assert!(!scalar_bits.is_null());
+
+    let len = 32usize;
+    let mut scalar_slice = [0u8; 32];
+    scalar_slice.copy_from_slice(&slice::from_raw_parts(scalar_bits, len)[..32]);
+
+    let scalar = Scalar::from_bits(scalar_slice);
```

Comment:
Btw, a nit: In Rust its typically discouraged to have a bunch of temporary `let`s that immediately get consumed, unless you need to introduce them to make rustfmt format in a nicer way, which is not the case here. So I'd "inline" `scalar`'s definition into its use.